### PR TITLE
MGMT-7941: Introduce dry mode to the agent

### PR DESCRIPTION
**Pull request by omertuc**
This commit introduces a dry run mode to the agent. The mode is
activated when the `DRY_ENABLE=true` environment is set (or when the
`--dry-run` flag is passed to the installer).

The dry run mode disables several destructive actions performed by the
agent. The purpose of the dry run mode is mostly allowing us to
run a lot of agents on the same machine, without causing harm to
that machine, but still communicating with the service. This is useful
when performing load testing to the service.

Other than disabling destructive actions, it also disables actions that
take too long due to networking / processing reasons.

See the diff for the exact changes, but here's a summary -
- Next step runner retry delay during dry run reduced from 1h to 1m. I believe
  1h was chosen to reduce load on production servers, and it's
  annoyingly long when dealing with crashes during load testing.
- Added journal `DRY_AGENT_ID` field to help separate between
  logs of multiple dry run agents running at the same time on the
  same machine
- When registering a host, the host ID defined by the `DRY_HOST_ID`
  environment will be used rather then the host ID retrieved from
  the hardware.
- `diagnoseSystem` is skipped, it's not necessary and consumes CPU
- Image availability is skipped, dummy results are returned instead
- Disk speed check - 1ms is returned immediately rather than doing
  destructive/slow disk checks
- `getBmcAddress`/`getBmcV6Address` are skipped, the default "not found"
  `0.0.0.0` or `::/0` are immediately returned
- `smartctl` calls are skipped, it's too slow and not useful, dummy
  hard-coded smartctl JSON is returned
- The first interface's MAC address is overridden with the MAC address
  specified in the `DRY_MAC_ADDRESS` env variable. This is useful for
  when you want multiple agents to run on the same machine, but present
  different mac addresses to the service, so they can be individually
  identiefied and separated by BMAC.
- NTP sync - returns an empty list immediately, it's too slow and not
  really necessary when doing a fake installation
- Added the `--cgroup` namespace to `nsenter`. The reasoning for that
  is explained in detail in a code comment in this commit's diff.
- Agent and next_step_runner will now halt when the file whose path
  is configured by DRY_FAKE_REBOOT_MARKER_PATH gets created. This file
  is used by the installer to signal that a "fake reboot" happened.

### CODE REVIEW

**Comment by osherdp** [comment]
there's still a failing unit-test, no?

**Comment by romfreiman** [comment]
do u cover dry run mode in ut?

**Comment by omertuc** [comment]
> do u cover dry run mode in ut?

Nope. Didn't think it was worth the added complexity for unit tests / effort as it is not used by customers, just in our internal load testing scenarios

File: Makefile [hunk 1]
```diff
@@ -57,7 +57,7 @@ ifeq ($(CI), true)
 endif
 
 unit-test:
-	$(MAKE) _test TEST_SCENARIO=unit TIMEOUT=30m TEST="$(or $(TEST),$(shell go list ./... | grep -v subsystem))" || (docker kill postgres && /bin/false)
+	$(MAKE) _test TEST_SCENARIO=unit TIMEOUT=30m TEST="$(or $(TEST),$(shell go list ./... | grep -v subsystem))" || (docker kill postgres || /bin/true)
```
[the remark this answers is not in the document]
Comment (omertuc) [reply Makefile:60]: It doesn't make sense

**Review by osherdp — APPROVED** [review]

**Pull request merged**